src/main.ts

```diff
@@ -93,7 +93,15 @@ export function injectEntry(
     return injectToArray(originalEntry, newEntry, options.entryOrder);
   }
 
-  if (originalEntry === Object(originalEntry)) {
+  if (typeof originalEntry === 'function') {
+    const callbackOriginEntry = originalEntry();
+    if (callbackOriginEntry instanceof Promise) { // can't handle Promise
+      return originalEntry;
+    }
+    return injectEntry(callbackOriginEntry, newEntry, options);
+  }
+
+  if (Object.prototype.toString.call(originalEntry).slice(8, -1) === 'Object') {

```

Comment:
Since you added the above check for a function, the original object check should still be valid.

Reply:
I think the original object check still be valid.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/toString